File: src/MrHotel/WebApp/src/app/types/model/reservation.ts
import { Guest } from "./guest"
import { Guid } from "./guid"
import { Room } from "./room"
import { Tag } from "./tag"

export type Reservation = {
  id: Guid,
  guest: Guest,
  room: Room,
  checkInDate: Date,
  checkOutDate: Date,
  price: number
}

File: src/MrHotel/Database/Entities/Reservations/Reservation.cs
namespace MrHotel.Database.Entities.Reservations;

using MrHotel.Database.Entities.Guests;
using MrHotel.Database.Entities.Rooms;

public partial class Reservation
{
    public required Guid Id { get; set; }

    public Guest Guest { get; set; } = null!;

    public Room Room { get; set; } = null!;

    public required DateTimeOffset CheckInDate { get; set; }

    public required DateTimeOffset CheckOutDate { get; set; }

    public bool CheckOutDone { get; set; }

    public float Price { get; set; }
}
